ansible/playbooks/adhoc/add_etcd_volume/add_etcd_volume.yml

```diff
@@ -0,0 +1,352 @@
+
+    # XXX RHEL7's python-jinja2 package is too old to have the 'eq'
+    #     test, or we could just use the 'selectattr' filter here.
+    - set_fact:
+        etcd_self: |
+          {%- set result = [] %}
+          {%- for member in etcd_member_data.members %}
+          {%-   if member.ID == etcd_member_data.header.member_id %}
+          {%-     set _ = result.append(member) %}
+          {%-   endif %}
+          {%- endfor %}
+          {{- result.pop() -}}
+
+    # XXX RHEL7's python-jinja2 package is too old to have the 'ne'
+    #     test, or we could just use the 'rejectattr' filter here.
+    - set_fact:
+        etcd_peers: |
+          {%- set result = [] %}
+          {%- for member in etcd_member_data.members %}
+          {%-   if member.ID != etcd_member_data.header.member_id %}
+          {%-     set _ = result.append(member) %}
+          {%-   endif %}
+          {%- endfor %}
+          {{- result -}}
+
+    - set_fact:
+        etcd_delegate_to: "{{ cli_clusterid }}-master-{{ etcd_peers[0].name }}"
+
+    - assert:
+        that: etcd_delegate_to in ansible_play_hosts
+
+    - name: Get container names for current master pods
+      command: "docker ps --format=\\{\\{.Names\\}\\}"
+      register: docker_ps
+
+    - name: Verify master static pods are running
+      assert:
+        that:
+        - "'k8s_POD_master-api-' in docker_ps.stdout"
+        - "'k8s_POD_master-controllers-' in docker_ps.stdout"
+        - "'k8s_POD_master-etcd-' in docker_ps.stdout"
+
+    - name: Create temporary directory for pod files
+      tempfile:
+        path: "{{ openshift_node_pods_dir | dirname }}"
+        state: directory
+        suffix: ".pods"
+      register: tempdir
+
+    - name: Stop master static pods
+      command: "mv {{ openshift_node_pods_dir }}/{{ item }} {{ tempdir.path }}/{{ item }}"
+      creates: "{{ tempdir.path }}/{{ item }}"
+      removes: "{{ openshift_node_pods_dir }}/{{ item }}"
+      with_items:
+      - apiserver.yaml
+      - controller.yaml
+      - etcd.yaml
+
+    - name: Wait for containers to stop
+      command: "docker ps --format=\\{\\{.Names\\}\\}"
+      register: docker_ps
+      until:
+      - "'k8s_POD_master-api-'         not in docker_ps.stdout"
+      - "'k8s_POD_master-controllers-' not in docker_ps.stdout"
+      - "'k8s_POD_master-etcd-'        not in docker_ps.stdout"
+      delay: 5
+      retries: 30
+
+    - name: Remove stopped member from etcd cluster
+      shell: ". /etc/profile.d/etcdctl.sh && etcdctl3 member remove {{ '%x' | format(etcd_self.ID) | quote }}"
+      delegate_to: "{{ etcd_delegate_to }}"
+
+    - name: Add stopped member to etcd cluster
+      shell: ". /etc/profile.d/etcdctl.sh && etcdctl3 member add {{ etcd_self.name | quote }} --peer-urls={{ etcd_self.peerURLs | join(',') | quote }}"
+      register: etcd_member_add
+      delegate_to: "{{ etcd_delegate_to }}"
+
```

Comment:
Do you have any example output from `etcdctl3 member add`? It might be helpful to put it here in comments, to give people an idea of what exactly is being parsed in etcd_conf_updates. Just in case we have to troubleshoot it later.

Reply:
Good idea, will add some sample output.